src/service/mainnet/index.ts

```diff
@@ -1,41 +1,94 @@
     this.subnetSmartContractInstance = new this.web3.eth.Contract(subnetContract.abi as AbiItem[], config.smartContractAddress);
+    this.mainnetAccount = this.web3.eth.accounts.privateKeyToAccount(config.accountPK);
   }
   
   /*
     A method to fetch the last subnet block that has been stored/audited in mainnet XDC
   **/
-  async getLastAuditedBlock(): Promise<XdcHeader> {
+  async getLastAuditedBlock(): Promise<MainnetBlock> {
     try {
       const blockHash = await this.subnetSmartContractInstance.methods.getLatestFinalizedBlock().call();
       const rlpEncodedBlockHeader = await this.subnetSmartContractInstance.methods.getHeader(blockHash).call();
-      return decodeInputToBlock(rlpEncodedBlockHeader);
+      const header =  decodeInputToBlock(rlpEncodedBlockHeader);
+      return {
+        mainnetBlockHash: blockHash,
+        mainnetHeader: header
+      };
     } catch (error) {
-      console.error("Error while trying to fetch the last audited subnet's block in XDC mainnet", error);
+      console.error("Error while trying to fetch the last audited subnet's block in XDC mainnet", {message: error.message});
       throw error;
     }
   }
   
-  async submitTransactions(txs: any): Promise<void> {
-    await sleep(2000);
-    console.log("[Mainnet] Submit new transactions into XDC mainnet");
+  /*
+    A method to fetch the last subnet block that has been stored/audited in mainnet XDC
+  **/
+    async getLastAuditedBlockHash(): Promise<string> {
+      try {
+        return await this.subnetSmartContractInstance.methods.getLatestFinalizedBlock().call();
+      } catch (error) {
+        console.error("Error while trying to fetch the last audited subnet's block hash in XDC mainnet's smart contract", {message: error.message});
+        throw error;
+      }
+    }
+  
+  async submitTxs(rlpEncodedHeaders: string[]): Promise<void> {
+    try {
+      for await (const h of rlpEncodedHeaders) {
+        const rlpBytes = base64ToUint8(h);
+        const transactionToBeSent = await this.subnetSmartContractInstance.methods.receiveHeader(rlpBytes);
+        const options = {
+          to: transactionToBeSent._parent._address,
+          data    : transactionToBeSent.encodeABI(),
+          gas     : await transactionToBeSent.estimateGas({from: this.mainnetAccount.address}),
+          gasPrice: TRANSACTION_GAS_NUMBER
+        };
+        const signed = await this.web3.eth.accounts.signTransaction(options, this.mainnetAccount.privateKey);
+        const receipt = await this.web3.eth.sendSignedTransaction(signed.rawTransaction);
+        console.info("Successfully submitted the subnet block as tx into mainnet", receipt);
+        await sleep(this.mainnetConfig.submitTransactionWaitingTime);
+      } 
+    } catch (error) {
+      console.error("Fail to submit transactions into mainnet", {message: error.message});
+      throw error;
+    }
+  }
+  
+  async submitTx(rlpEncodedString: string): Promise<void> {
```

Comment:
This method is for debugging purpose. will delete after we fix the block 1 issue